README.md

```diff
@@ -1,2 +1,178 @@
+    <active>true</active>
+    <protocol>http</protocol>
+    <host>someHost</host>
+    <port>8080</port>
+    <username>UserName</username>
+    <password>Password</password>
+    <nonProxyHosts>*.google.com|*jitpack.io</nonProxyHosts>
+</proxy>
+```
+
+# How to consume the dependency
+
+   ```
+   <dependency>
+      <groupId>com.github.znsio</groupId>
+      <artifactId>ReportPortalIntegration</artifactId>
+      <version>x.x.x</version>
+   </dependency>
+   ```
+
+# How to configure ReportPortal in your automation framework
+
+1. **Listener Configuration**
+    1. **Approach-1 Listener in TestNG XML file:** Add the following line to configure ReportPortal listener in the xml
+       file where you have configured your testng suite
+        1. **Without overloading properties**
+           ```
+           <listener class-name="com.epam.reportportal.testng.ReportPortalTestNGListener"/>
+           ```
+        2. **With overloading properties**
+           ```
+           <listener class-name="com.znsio.rpi.listener.ReportPortalListener"/>
+           ```
+    2. **Approach-2 Listener in Maven Surefire Plugin:** Add the following snippet to configure ReportPortal listener in
+       the `pom.xml` file where you have configured your `maven-surefire-plugin`
+        1. **Without overloading properties**
+           ```
+           <property>
+              <name>listener</name>
+              <value>com.epam.reportportal.testng.ReportPortalTestNGListener</value>
+           </property>
+           ```
+        2. **With overloading properties**
+           ```
+           <property>
+              <name>listener</name>
+              <value>com.znsio.rpi.listener.ReportPortalListener</value>
+           </property>
+           ```
+    2. **Approach-3:** Use the following snippet to configure ReportPortal listener when creating TestNG file
+       programmatically
+        1. **Without overloading properties**
+           ```
+           ReportPortalTestNGListener reportPortalTestNGListener = new ReportPortalTestNGListener();
+           testNG.addListener(reportPortalTestNGListener);
+           ```
+        2. **With overloading properties**
+           ```
+           ReportPortalListener reportPortalListener = new ReportPortalListener();
+           testNG.addListener(reportPortalListener);
+           ```
+    3. For more details on listener configuration,
+       refer [reportportal-agent-java-testNG](https://github.com/reportportal/agent-java-testNG)
+2. **Properties file:** Create a `reportportal.properties` file either in `src/main/resources/` or
+   in `src/test/resources` directory in your automation framework with the following mandatory attributes
+   ```
+   rp.endpoint = <Your Endpoint>
+   rp.uuid = <Your UDID>
+   rp.launch = <Your Launch Name>
+   rp.project = <Your Project Name>
+   ```
+3. **ReportPortal Attributes and Properties Overlaoding:** Use the `ReportPortalPropertiesOverloader` class defined
```

Comment:
Why would the consumer of ReportPortalIntegration need to know about ReportPortalPropertiesOverloader?

Reply:
The details are given merely for understanding on the implementation of how the properties are being overloaded at run time. I've removed the usage of word "Use" from each statement to avoid confusion that this is something consumer can access.

Reply:
Removed it from here as it was under "How to configure ReportPortal in your automation framework" section and we don't allow any overriding from consumer side on `ReportPortalPropertiesOverloader` class

Reply:
Removed it from here as it was under "How to configure ReportPortal in your automation framework" section and we don't allow any overriding from consumer side on `ReportPortalPropertiesOverloader` class